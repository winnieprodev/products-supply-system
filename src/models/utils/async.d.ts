export interface Error {
  message: string;
}

export interface Async<T> {
  isLoading: boolean;
  error: Error | undefined;
  data: T | undefined;
}
