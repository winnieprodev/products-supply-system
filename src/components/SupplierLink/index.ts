export { default as SupplierLink } from "./SupplierLink";
