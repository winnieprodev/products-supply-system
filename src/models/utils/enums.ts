export enum WrapperMode {
  full = "full",
  singleLine = "singleLine",
  tiny = "tiny",
}
