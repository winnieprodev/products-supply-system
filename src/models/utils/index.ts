export * from "./async.d";
export * from "./enums";
