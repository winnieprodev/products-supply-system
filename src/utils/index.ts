export * from "./routeUtils";
export * from "./supplierUtils";
export * from "./testUtils";
