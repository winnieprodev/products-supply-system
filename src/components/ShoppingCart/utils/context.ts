import { Async, Cart, ProductCartInfo } from "models";
import React from "react";
import { mockAsync, mockCart } from "utils";

interface ShoppingCartContextProps {
  cartId: Async<number>;
  cart: Async<Cart> | undefined;
  setCart: (cart: Async<Cart> | undefined) => void;
  setProduct: (product: ProductCartInfo) => void;
  removeProduct: (productId: string) => void;
}

export const EMPTY_CART = { products: [] };
const mockFunction = () => {};

export const ShoppingCartContext = React.createContext<ShoppingCartContextProps>(
  {
    cartId: mockAsync(0),
    cart: mockAsync(mockCart(0)),
    setCart: mockFunction,
    setProduct: mockFunction,
    removeProduct: mockFunction,
  },
);
