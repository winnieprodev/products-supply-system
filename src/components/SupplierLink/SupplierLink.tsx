import { Supplier } from "models";
import React from "react";
import { Link } from "react-router-dom";
import ROUTES, { SupplierType } from "routes";

interface OwnProps {
  item: Supplier;
  supplierType: SupplierType;
  outletId: string;
}

export const SupplierLink: React.FC<OwnProps> = ({
  item,
  supplierType,
  outletId,
}) => (
  <div>
    Supplier:{" "}
    <Link
      to={ROUTES.SUPPLIER_DETAILS.use({
        outletId,
        supplierType,
        supplierId: item.id,
      })}
    >
      {item.title}
    </Link>
  </div>
);

export default React.memo(SupplierLink);
