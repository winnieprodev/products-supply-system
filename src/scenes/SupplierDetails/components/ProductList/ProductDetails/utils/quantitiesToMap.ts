import { Map } from "immutable";
import { CartQuantity, QuantityType } from "models";

export const quantitiesToMap = (quantities: CartQuantity[]) => {
  return Map<QuantityType, number>(
    quantities.map(({ type, amount }) => [type, amount] as const) as Iterable<
      [QuantityType, number]
    >,
  );
};
