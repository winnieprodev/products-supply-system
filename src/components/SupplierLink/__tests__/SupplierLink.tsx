import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { mockSupplier } from "utils";

import SupplierLink from "../SupplierLink";

describe("SupplierLink", () => {
  describe("Snapshot tests", () => {
    test("it matches the snapshot", () => {
      const { container } = render(
        <BrowserRouter>
          <SupplierLink
            outletId="id"
            item={mockSupplier("title test")}
            supplierType="local"
          />
        </BrowserRouter>,
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
