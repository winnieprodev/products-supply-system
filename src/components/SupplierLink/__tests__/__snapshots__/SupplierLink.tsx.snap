// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupplierLink Snapshot tests it matches the snapshot 1`] = `
<div>
  Supplier:
   
  <a
    href="/browse/id/local/title-test/"
  >
    title test
  </a>
</div>
`;
